src/gui/sharedialog.cpp

```diff
@@ -0,0 +1,308 @@
+#include "sharedialog.h"
+#include "ui_sharedialog.h"
+#include "networkjobs.h"
+#include "account.h"
+#include "json.h"
+#include "folderman.h"
+#include <QBuffer>
+#include <QMovie>
+
+namespace {
+    int SHARETYPE_PUBLIC = 3;
+}
+
+namespace OCC {
+
+ShareDialog::ShareDialog(QWidget *parent) :
+    QDialog(parent),
+    _ui(new Ui::ShareDialog)
+{
+    setAttribute(Qt::WA_DeleteOnClose);
+    _ui->setupUi(this);
+    _ui->pushButton_copy->setIcon(QIcon::fromTheme("edit-copy"));
+    layout()->setSizeConstraint(QLayout::SetFixedSize);
+    QMovie *movie = new QMovie("/home/azelphur/ownCloud-share-tools/loading-icon.gif");
```

Comment:
Yeah I spotted that as well. Do we already have a default spinner in the resources? And do we even want one?